### ISSUE TITLE
prov/rxm: fi_readmsg() failed, rc: -113(No route to host)

### ISSUE THREAD

**Issue by yulujia** [ofiwg/libfabric#5507]
I ran a test we have using the tcp provider, after the clients connect/disconnect for a few cycles, the server side starts to print out a lot of messages like the following, and test gets stuck:

using libfabfic commit:

```2d2b11e54 - Merge pull request #5498 from SystemFabricWorks/xrc_sidr_exchange (5 hours ago) <Sean Hefty>```
```12/17-16:13:07.52 wolf-23 DAOS[46648/46661] common DBUG src/common/misc.c:44 daos_sgl_init() alloc(calloc) 'sgl->sg_iovs': 24 at 0x7f1066fc4160.                                                                   
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] rpc  DBUG src/cart/crt_rpc.c:473 crt_req_addref(0x7f1066fc3670)  [opc=0x401000b rank:xid=0:0xe3a tag=6] addref to 2.                                                   
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   DBUG src/cart/crt_hg.c:1614 crt_hg_bulk_transfer() alloc(calloc) 'bulk_cbinfo': 24 at 0x7f1066fc4260.                                                             
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   DBUG src/cart/crt_hg.c:1617 crt_hg_bulk_transfer() alloc(calloc) 'bulk_desc_dup': 56 at 0x7f1066fc4280.                                                           
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   ERR  # NA -- Error -- /home/user/codes/daos-fresh-cart-764-tcp/_build.external/mercury/src/na/na_ofi.c:4241                                                       
 # na_ofi_get(): fi_readmsg() failed, rc: -113(No route to host)                                                                                                                                                   
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   ERR  # HG -- Error -- /home/user/codes/daos-fresh-cart-764-tcp/_build.external/mercury/src/mercury_bulk.c:804                                                     
 # hg_bulk_transfer_pieces(): Could not transfer data                                                                                                                                                              
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   ERR  # HG -- Error -- /home/user/codes/daos-fresh-cart-764-tcp/_build.external/mercury/src/mercury_bulk.c:968                                                     
 # hg_bulk_transfer(): Could not transfer data pieces                                                                                                                                                              
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   ERR  # HG -- Error -- /home/user/codes/daos-fresh-cart-764-tcp/_build.external/mercury/src/mercury_bulk.c:2017                                                    
 # HG_Bulk_transfer_id(): Could not transfer data                                                                                                                                                                  
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   ERR  src/cart/crt_hg.c:1655 crt_hg_bulk_transfer() HG_Bulk_(bind)transfer failed, hg_ret: 1.                                                                      
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   DBUG src/cart/crt_hg.c:1656 crt_hg_bulk_transfer() free 'bulk_cbinfo' at 0x7f1066fc4260.                                                                          
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] hg   DBUG src/cart/crt_hg.c:1657 crt_hg_bulk_transfer() free 'bulk_desc_dup' at 0x7f1066fc4280.                                                                        
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] bulk ERR  src/cart/crt_bulk.c:231 crt_bulk_bind_transfer() crt_hg_bulk_transfer failed, rc: -1020.                                                                     
12/17-16:13:07.52 wolf-23 DAOS[46648/46661] object ERR  src/object/srv_obj.c:315 ds_bulk_transfer() crt_bulk_transfer 0 error (-1020).```

**Comment by shefty** [comment]
Is there a way I can reproduce this problem?

Can you see if you can reproduce the issue using an updated fi_rdm test, with the -C option?  That will connect multiple EPs to the server side.

**Comment by shefty** [comment]
Problem was recently fixed in master and v1.11.x branches.
